test/estimate-gas.test.js

```diff
@@ -0,0 +1,138 @@
+            recordsForCSV.push( { scope: 'atonomi', task: 'add mfgr', gas: gas } )
+        })
+        it('   ^= to Add IRN Nodes', async () => {
+            var gas = await ctx.contracts.atonomi.addNetworkMember.estimateGas(ctx.actors.admin, false, false, true, '', {from: ctx.actors.owner})
+            console.log('      ✓ ' + gas + ' wei')
+            recordsForCSV.push( { scope: 'atonomi', task: 'add irn node', gas: gas } )
+        })
+    })
+
+    describe('manufacturer costs', () => {
+        it('   ^= to Register Device', async () => {
+            await ctx.contracts.token.contract.approve(ctx.contracts.atonomi.address, regFee, { from: ctx.actors.mfg }) //results in estimate of next line = 165980
+            //await ctx.contracts.token.contract.approve(ctx.contracts.atonomi.address, (regFee + actFee), { from: ctx.actors.mfg }) // results in estimate of next line = 180980
+            const gas = await ctx.contracts.atonomi.registerDevice.estimateGas(deviceIdHash, deviceType, devicePublicKey, {from: ctx.actors.mfg})
+            console.log('      ✓ ' + gas + ' wei')
+            recordsForCSV.push( { scope: 'mfgr', task: 'register device', gas: gas } )
+        })
+        it('   ^= to Register and Activate Device', async () => {
+            await ctx.contracts.token.contract.approve(ctx.contracts.atonomi.address, (regFee + actFee), { from: ctx.actors.mfg })
+            var gas = await ctx.contracts.atonomi.registerAndActivateDevice.estimateGas(deviceId, deviceType, devicePublicKey, {from: ctx.actors.mfg})
+            console.log('      ✓ ' + gas + ' wei')
+            recordsForCSV.push( { scope: 'mfgr', task: 'reg & act device', gas: gas } )
+        })
+        it('   ^= to Withdraw Tokens', async () => {
+            await ctx.contracts.token.contract.approve(ctx.contracts.atonomi.address, regFee + actFee, {from: ctx.actors.mfg})
+            await ctx.contracts.atonomi.registerAndActivateDevice(deviceId, deviceType, devicePublicKey, {from: ctx.actors.mfg})
+            await ctx.contracts.atonomi.updateReputationScore(deviceId, score, {from: ctx.actors.irnNode})
+            const testWithdraw = { account: ctx.actors.mfg, expectedTokens: mfgReward }
+            const from = testWithdraw.account
+            var gas  = await ctx.contracts.atonomi.withdrawTokens.estimateGas({ from: from })
+            console.log('      ✓ ' + gas + ' wei')
+            recordsForCSV.push( { scope: 'mfgr', task: 'withdraw tokens', gas: gas } )
+        })
+    })
+
+    describe('owner costs', () => {
+        it('   ^= to Activate Device', async() => {
+            await ctx.contracts.token.contract.approve(ctx.contracts.atonomi.address, regFee, { from: ctx.actors.mfg })
+            await ctx.contracts.atonomi.registerDevice(deviceIdHash, deviceType, devicePublicKey, {from: ctx.actors.mfg})
+            await ctx.contracts.token.contract.approve(ctx.contracts.atonomi.address, actFee, { from: ctx.actors.deviceOwner })
```

Comment:
add line item for approve cost as well (since they are paying this in metamask)

Reply:
or we could just add it to the total of register cost

Reply:
i added it to the total, and denoted that in the description: ```^= to Activate Device (incl. Approval)```